### PR TITLE
make `eventSystem/event` alpaka compatible

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- `CudaEvent.hpp`
  - remove plain cuda types
  - add alpaka types for events and streams
- `kernelEvents.hpp`
  - add alpaka types to start a kernel
  - keep the old interface of `__cudaKernel`
  - each kernel needs to define `::dim` to be compatible with `__cudaKernel`

Merge after:
- [x] issue #4

### CODE REVIEW

File: src/libPMacc/include/eventSystem/events/CudaEvent.hpp [hunk 1]
```diff
@@ -93,12 +97,19 @@ class CudaEvent
      *
      * @return native cuda event
      */
-    cudaEvent_t operator*() const
+    const AlpakaEvent& operator*() const
+    {
+        assert(isValid);
+        return *event.get();
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/events/CudaEvent.hpp]: The use of '*sharedPtr' should be enought here.

File: src/libPMacc/include/eventSystem/events/CudaEvent.hpp [hunk 2]
```diff
@@ -93,12 +97,19 @@ class CudaEvent
...
+    }
+
+    AlpakaEvent& operator*()
     {
         assert(isValid);
-        return event;
+        return *event.get();
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/events/CudaEvent.hpp]: The use of '*sharedPtr' should be enought here

File: src/libPMacc/include/eventSystem/events/CudaEvent.hpp [hunk 3]
```diff
@@ -107,7 +118,7 @@ class CudaEvent
     bool isFinished() const
     {
         assert(isValid);
-        return cudaEventQuery(event) == cudaSuccess;
+        return alpaka::event::test(*event.get());
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/events/CudaEvent.hpp]: The use of '*sharedPtr' should be enought here

File: src/libPMacc/include/eventSystem/events/CudaEvent.hpp [hunk 4]
```diff
@@ -116,10 +127,16 @@ class CudaEvent
      *
      * @return native cuda stream
      */
-    cudaStream_t getStream() const
+    const alpaka::AccStream& getStream() const
+    {
+        assert(isRecorded);
+        return *stream.get();
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/events/CudaEvent.hpp]: The use of '*sharedPtr' should be enought here

File: src/libPMacc/include/eventSystem/events/CudaEvent.hpp [hunk 5]
```diff
@@ -116,10 +127,16 @@ class CudaEvent
...
+    }
+
+    alpaka::AccStream& getStream()
     {
         assert(isRecorded);
-        return stream;
+        return *stream.get();
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/events/CudaEvent.hpp]: The use of '*sharedPtr' should be enought here

File: src/libPMacc/include/eventSystem/events/CudaEvent.hpp [hunk 6]
```diff
@@ -127,13 +144,13 @@ class CudaEvent
...
      */
-    void recordEvent(cudaStream_t stream)
+    void recordEvent(const alpaka::AccStream& stream)
     {
         /* disallow double recording */
         assert(isRecorded==false);
         isRecorded = true;
         this->stream = stream;
-        CUDA_CHECK(cudaEventRecord(event, stream));
+        alpaka::stream::enqueue(*stream.get(), *event.get());
```
Comment (erikzenker) [inline src/libPMacc/include/eventSystem/events/CudaEvent.hpp]: Which stream is used here. Maybe we should not shadow the stream variable here.

The use of '*sharedPtr' should be enought here

**Comment by psychocoderHPC** [comment]
ready for merge, I will put the changes in a separate pull request

**Comment by erikzenker** [comment]
Good Work

**Pull request merged**